dist/pythonlibs/testrunner/pytest.py

```diff
@@ -20,17 +20,37 @@
 import subprocess
 import re
 import time
+import traceback
 
 import pexpect
 import pytest
 import testrunner
 
-
 DEFAULT_TIMEOUT = 10
 TEST_LOG_CONSOLE = bool(int(os.environ.get('TEST_LOG_CONSOLE', '1')))
 TEST_TIMEOUT_ATTR = 'TIMEOUT'
 
 
+class CustomSpawn(pexpect.spawn):
+    """Convenient subclass to better catch pexpect timeout and eof errors."""
+
+    def expect(self, pattern, timeout=-1, searchwindowsize=-1, async_=False,
+               **kw):
+        # pylint:disable=arguments-differ
+        try:
+            super(CustomSpawn, self).expect(pattern, timeout=timeout,
```

Comment:
Question added to the `async_` case, I know it has now been changed as `async` as become a keyboard but this is also pexpect version specific. Upgrading to the new api now is not a bad thing but it would then also need to be handled when saying `async=async_`.
https://pexpect.readthedocs.io/en/stable/history.html#version-4-3

Or a solution could be to stay currently expect specific and only repeat the required arguments for `expect` and put everything possible to `*args` and `**kw`.

Reply:
+1, I changed that to only use **kw